app/Libraries/BeatmapsetDiscussionPostsBundle.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace App\Libraries;
+
+use App\Models\BeatmapDiscussionPost;
+use App\Models\Beatmapset;
+use App\Models\User;
+use App\Traits\Memoizes;
+use App\Transformers\BeatmapDiscussionPostTransformer;
+use App\Transformers\BeatmapsetCompactTransformer;
+use App\Transformers\UserCompactTransformer;
+use Illuminate\Pagination\Paginator;
+
+class BeatmapsetDiscussionPostsBundle extends BeatmapsetDiscussionsBundleBase
+{
+    use Memoizes;
+
+    public function getData()
+    {
+        return $this->getPosts();
+    }
+
+    public function toArray()
+    {
+        return [
+            'beatmapsets' => json_collection($this->getBeatmapsets(), new BeatmapsetCompactTransformer()),
+            'cursor' => $this->getCursor(),
+            'posts' => json_collection($this->getPosts(), new BeatmapDiscussionPostTransformer(), ['beatmaps', 'users']),
+            'users' => json_collection($this->getUsers(), new UserCompactTransformer(), ['groups']),
+        ];
+    }
+
+    private function getBeatmapsets()
+    {
+        return $this->memoize(__FUNCTION__, function () {
+            $beatmapsetIds = $this->getPosts()->pluck('beatmapset.beatmapset_id')->unique()->values();
+
+            return Beatmapset::whereIn('beatmapset_id', $beatmapsetIds)->get();
+        });
+    }
+
+    private function getPosts()
+    {
+        return $this->memoize(__FUNCTION__, function () {
+            $this->search = BeatmapDiscussionPost::search($this->params);
```

Comment:
That'd still be weird because `getSearch` would be including a query builder that could be modified before `getPosts` actually calls it, instead of returning the just search that `getPosts()` used.

Reply:
There doesn't seem to be a need for query to be returned though 🤔

Reply:
looks like a better solution is to split off the search function to params cleaner and query builder